### PR TITLE
Add Azure Storage Backend

### LINKED ISSUE #62: Add Azure Storage Backend
Create a new backend to support Microsoft Azure Object storage.

### PR DESCRIPTION
**Pull request by spacentropy**
Fixes #62

## Proposed Changes
- Azure Storage backend

### Description

I just added Azure blob storage backend.

### Checklist

- [x] Read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] Read the [**CODE OF CONDUCT**](CODE_OF_CONDUCT.md) document.
- [x] Add tests to cover changes.
- [x] Ensure your code follows the code style of this project.
- [x] Ensure CI and all other PR checks are green OR
    - [x] Code compiles correctly.
    - [x] Created tests which fail without the change (if possible).
    - [x] All new and existing tests passed.
- [x] Add your changes to `Unreleased` section of [CHANGELOG](CHANGELOG.md).
- [x] Improve and update the [README](README.md) (if necessary).
- [x] Ensure [documentation](./DOCS.md) is up-to-date. The same file will be updated in [plugin index](https://github.com/drone/drone-plugin-index/blob/master/content/meltwater/drone-cache/index.md) when your PR is accepted, so it will be available for end-users at http://plugins.drone.io.

### CODE REVIEW

**Comment by kakkoyun** [comment]
@spacentropy Wow. This looks great at a first glimpse. Thank you very much.
I'll thoroughly review this as soon as I can.

**Comment by spacentropy** [comment]
Thanks. Let me green the build first, there are multiple linter errors :( I didn't use golangci-lint. Sorry for that, will fix now.

File: DOCS.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 date: 2019-03-19T00:00:00+00:00
 title: Drone Cache
 author: meltwater
-tags: [ cache, amazon, aws, s3, volume ]
+tags: [ cache, amazon, aws, s3, azure, volume ]
```
Comment (kakkoyun) [inline DOCS.md]:
```suggestion
tags: [ cache, amazon, aws, s3, azure, blob, volume, storage ]
```

File: DOCS.md [hunk 2]
```diff
@@ -354,6 +354,15 @@ bucket
 region
 : AWS bucket region. (`us-east-1`, `eu-west-1`, ...)
 
+account_name
```
Comment (kakkoyun) [inline DOCS.md]: Let's add prefixes to these. We don't add prefixes to AWS ones to be back-compatible.

Comment (kakkoyun) [reply DOCS.md]: You actually add the `azure` prefix we just need to update here in docs.

File: cache/backend/azure.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	bodyStream := downloadResponse.Body(azblob.RetryReaderOptions{MaxRetryRequests: 4})
+
+	return bodyStream, nil
+}
+
+// Put uploads the contents of the io.ReadSeeker
+func (c *azureBackend) Put(p string, src io.ReadSeeker) error {
+	blobURL := c.containerURL.NewBlockBlobURL(p)
+
+	fmt.Printf("uploading the file with blob name: %s\n", p)
```
Comment (kakkoyun) [inline cache/backend/azure.go]: You should use log, or remove this one. If you don't have logger available.

**Pull request merged**